site/source/docs/porting/exceptions.rst

```diff
@@ -32,19 +23,54 @@ For example, build the following program with ``emcc -fexceptions``:
       return 0;
     }
 
-Running it will print ``throw...`` and then ``catch!``.
+The first ``throw`` will abort the program and you'll see something like this in the output:
+
+.. code-block:: text
+
+  throw...  
+  exception thrown: 5246024 - Exception catching is disabled, this exception cannot be caught. Compile with -s NO_DISABLE_EXCEPTION_CATCHING or -s EXCEPTION_CATCHING_ALLOWED=[..] to catch.
+
+If you want to opt-in, you have two following options.
+
+JavaScript-based exception support
+##################################
+
+First, you can enable exceptions via Emscripten's JavaScript-based support.
+
+To enable it, pass ``-fexceptions`` at both compile time and link time.
+
+When you rebuild the example above with this flag, the output will change to:
+
+.. code-block:: text
+
+  throw...
+  catch!
+
+Note that this option has relatively high overhead, but it will work on all
+engines with WebAssembly support. You can reduce the overhead by specifying a
+list of allowed functions in which exceptions are enabled, see the
+``EXCEPTION_CATCHING_ALLOWED`` setting.
+
+WebAssembly exception handling proposal
+#######################################
+
+Alternatively, you can opt-in to the `native WebAssembly exception handling
+<https://github.com/WebAssembly/exception-handling/blob/master/proposals/exception-handling/Exceptions.md>`_.
+
+To enable it, pass ``-fwasm-exceptions`` at both compile time and link time.
+
+Rebuilding the example with this flag will result in the same output as with
+``-fexceptions`` above:
 
-Overhead
-########
+.. code-block:: text
 
-Until wasm gets direct support for exceptions, Emscripten supports them using
-JavaScript for throwing and catching. As a result, every time we might need to
-catch an exception we call out into JavaScript to a try-catch there. This
-increases code size and adds significant overhead.
+  throw...
+  catch!
 
-You can reduce the overhead by specifying a list of allowed functions in
-which exceptions are enabled, see the ``EXCEPTION_CATCHING_ALLOWED`` setting.
+This option leverages a new feature that brings built-in instructions for
+throwing and catching exceptions to WebAssembly.
 
-In the future direct support for exceptions in wasm will avoid that overhead.
-Usage will be as mentioned earlier, but with ``-fwasm-exceptions`` instead of
-``-fexceptions``.
+As a result, it can reduce amount of custom glue and performance overhead
+compared to the JavaScript-based implementation, but it's also highly
```

Comment:
I think it's better than "highly experimental" at this point. Since it's so close to done (and works pretty well in our real-world tests) maybe we could just say "but it's also a brand-new feature and not yet supported..." Or if you want to make the language more scary than that, you could call it a "bleeding-edge" feature. But at this point I think we want to be encouraging rather than discouraging people to try it.

Reply:
Fair enough, I wasn't sure how stable it's considered on tooling side.